### PR TITLE
Feature/order/confirm dialog

### PR DESCRIPTION
**Pull request by MaxSukhovetskyi**

### CODE REVIEW

File: easyrest/models/order.py [hunk 1]
```diff
@@ -98,7 +103,91 @@ def count_total(self):
         total = 0
         for item in self.items:
             q = item.quantity
-            p_per_item = item.food.price if item.food.price is not None else 3.50
+            p_per_item = item.food.price if item.food.price is not None else 0
```
Comment (EvanAntoni) [inline easyrest/models/order.py]: Here probably better to set the default price in the `menu_item.py`

File: easyrest/models/user.py [hunk 2]
```diff
@@ -6,13 +6,15 @@
     Text,
     Date,
     ForeignKey,
-    Boolean
+    Boolean,
+    inspect
```
Comment (EvanAntoni) [inline easyrest/models/user.py]: This thing is not used.

File: easyrest/models/user.py [hunk 3]
```diff
@@ -6,13 +6,15 @@
...
 )
 from sqlalchemy.orm import relationship
 from passlib.hash import pbkdf2_sha256
 
 from .meta import Base
 from validator import validation
+from .user_role import UserRole
```
Comment (EvanAntoni) [inline easyrest/models/user.py]: As well as this. Not used.

File: easyrest/routes.py [hunk 4]
```diff
@@ -53,3 +53,6 @@ def includeme(config):
                      request_method=["POST", "GET", "PUT", "DELETE"])
     config.add_route('order_status', 'order/{order_id:\d+}/status',
                      request_method=["GET", "PUT"])
+
+    config.add_route('test', 'test',
+                     request_method=["GET"])
```
Comment (EvanAntoni) [inline easyrest/routes.py]: Test route, maybe it needs to be removed

File: easyrest/views/test.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+
+@view_config(route_name='test', renderer='json', request_method='GET')
+def test(request):
+    waiter = request.dbsession.query(User).filter_by(id=17).first()
+    waiter2 = request.dbsession.query(User).filter_by(id=10).first()
+    order = request.dbsession.query(Order).order_by(Order.id).first()
+    # order.waiter = waiter2
+    # request.dbsession.flush()
+    return order.as_dict()
```
Comment (EvanAntoni) [inline easyrest/views/test.py]: Test view

File: easyrest/views/order_controller.py [hunk 6]
```diff
@@ -5,6 +5,7 @@
 
 from pyramid.view import view_config
 from pyramid.httpexceptions import HTTPNotFound, HTTPForbidden, HTTPBadRequest
+from sqlalchemy.sql.expression import text
```
Comment (EvanAntoni) [inline easyrest/views/order_controller.py:8]: This thing is not used too.

**Review by RazSKg — APPROVED** [review]

File: easyrest/views/order_controller.py [hunk 7]
```diff
@@ -17,9 +18,24 @@
 from ..exceptions import ValidationError
 
 
-def get_order(request, order_id):
-    order = request.dbsession.query(Order).filter(
-        Order.id == order_id, Order.user_id == request.token.user.id).first()
+def get_order(request, order_id, filterList=False):
+    """'Smart' get order. If you want to specify some constrains
```
Comment (EvanAntoni) [inline easyrest/views/order_controller.py:23]: `filterList` maybe better with the underscore

File: easyrest/views/order_controller.py [hunk 8]
```diff
@@ -17,9 +18,24 @@
...
+        {
+            "field_name": value
+            ...
+        }
+    """
+    if filterList:
+        order = request.dbsession.query(Order).filter(Order.id == order_id)
+        for constrain in filterList:
+            order = order.filter(constrain)
+        order = order.first()
```
Comment (EvanAntoni) [inline easyrest/views/order_controller.py:35]: `filterList` underscore

File: easyrest/views/order_controller.py [hunk 9]
```diff
@@ -345,75 +364,88 @@ def change_status(request):
 
     try:
         json = request.json_body
-    except ValueError as e:
+    except ValueError:
         raise HTTPBadRequest("Not valid json")
```
Comment (EvanAntoni) [inline easyrest/views/order_controller.py]: The module validator.py already has a JSON format check method. You can use it in the future if you want.

**Pull request merged**